client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.controller.js

```diff
@@ -0,0 +1,376 @@
+class CloudProjectComputeInfrastructureVirtualMachineAddCtrl {
+    constructor ($q, $stateParams,
+                 CloudFlavorService, CloudImageService, CloudNavigation, CloudProjectVirtualMachineAddService,
+                 OvhCloudPriceHelper, OvhApiCloudProjectFlavor, OvhApiCloudProjectImage, OvhApiCloudProjectNetworkPrivate,
+                 OvhApiCloudProjectNetworkPublic, OvhApiCloudProjectQuota, OvhApiCloudProjectRegion, OvhApiCloudProjectSnapshot, OvhApiCloudProjectSshKey,
+                 CurrencyService, RegionService, ServiceHelper, ovhDocUrl) {
+        this.$q = $q;
+        this.$stateParams = $stateParams;
+        this.CloudFlavorService = CloudFlavorService;
+        this.CloudImageService = CloudImageService;
+        this.CloudNavigation = CloudNavigation;
+        this.OvhCloudPriceHelper = OvhCloudPriceHelper;
+        this.OvhApiCloudProjectFlavor = OvhApiCloudProjectFlavor;
+        this.OvhApiCloudProjectImage = OvhApiCloudProjectImage;
+        this.OvhApiCloudProjectNetworkPrivate = OvhApiCloudProjectNetworkPrivate;
+        this.OvhApiCloudProjectNetworkPublic = OvhApiCloudProjectNetworkPublic;
+        this.OvhApiCloudProjectQuota = OvhApiCloudProjectQuota;
+        this.OvhApiCloudProjectRegion = OvhApiCloudProjectRegion;
+        this.OvhApiCloudProjectSnapshot = OvhApiCloudProjectSnapshot;
+        this.OvhApiCloudProjectSshKey = OvhApiCloudProjectSshKey;
+        this.CurrencyService = CurrencyService;
+        this.RegionService = RegionService;
+        this.ServiceHelper = ServiceHelper;
+        this.VirtualMachineAddService = CloudProjectVirtualMachineAddService;
+        this.ovhDocUrl = ovhDocUrl;
+    }
+
+    $onInit () {
+        this.serviceName = this.$stateParams.projectId;
+        this.previousState = this.CloudNavigation.getPreviousState();
+        this.currentCurrency = this.CurrencyService.getCurrentCurrency();
+        this.loaders = {
+            adding: false
+        };
+        this.model = {
+            billingPeriod: null,
+            flavor: null,
+            imageType: null,
+            name: "",
+            networkId: "",
+            number: 0,
+            region: null,
+            sshKey: null,
+            userData: null
+        };
+        this.enums = {
+            billingPeriods: ["monthly", "hourly"],
+            flavorsTypes: [],
+            imagesTypes: []
+        };
+        this.isNameUpdated = false;
+        this.isPostScriptEnabled = false;
+        this.newSshKey = {
+            name: null,
+            publicKey: null
+        };
+        this.state = {
+            hasVRack: false
+        };
+        this.urls = {};
+    }
+
+    initProject () {
+        // Get prices in background
```

Comment:
Just try to remove as much comments as you can.. they are useless.

Reply:
done